src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,127 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
+    public void growIfArrayIsFull() {
+        if (size == elements.length) {
+            Object[] newArray = new Object[Math.round(elements.length * 1.5f)];
+            for (int i = 0; i < elements.length; i++) {
+                newArray[i] = elements[i];
+            }
+            elements = newArray;
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        growIfArrayIsFull();
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index didn't exist");
+        }
+        if (elements[index] == null) {
+            add(value);
+        } else {
+            growIfArrayIsFull();
+            size++;
+            for (int i = size - 1; i > index; i--) {
+                elements[i] = elements[i - 1];
+            }
+            elements[index] = value;
+        }
```

Comment:
The check for the null element at the index is not necessary since the logic for adding an element should be consistent regardless of the current element at the index. The shifting logic should always happen when inserting at a specific index.